### PR TITLE
[RTD-193] Creation of endpoints to store and retrieve data

### PR DESCRIPTION
**Pull request by and-mora**
#### Description

This PR proposes to implement and expose the endpoints necessary to store and retrieve the sender code and the relative key associated.

#### List of Changes

- Add endpoints in the rest controller
- Implement service
- Add unit tests for controller and service

#### Motivation and Context

The sender-auth must expose one endpoint to store the mapping between api key and sender code, and one endpoint to retrieve the sender code from the api key.

#### How Has This Been Tested?

- [X] Unit Test Suite
- [ ] Integration Test Suite
- [x] Api Tests
- [ ] Load Tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/DefaultRestController.java [hunk 1]
```diff
@@ -15,8 +20,16 @@
 @Validated
 public interface DefaultRestController {
```
Comment (GiovanniMancini) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/DefaultRestController.java:21]: @and-mora Wouldn't it be nice to have a more meaningful name here?

File: pom.xml [hunk 2]
```diff
@@ -21,6 +21,10 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
+		<dependency>
```
Comment (GiovanniMancini) [inline pom.xml:24]: @and-mora Who is using the health-check included in actuator?

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/RestControllerImpl.java [hunk 3]
```diff
@@ -8,10 +10,20 @@
...
 @ResponseBody
 @Slf4j
+@RequiredArgsConstructor
 public class RestControllerImpl implements DefaultRestController {
 
+  private final SenderAuthService authService;
+
+  @Override
+  public String getSenderCode(String apiKey) {
+    log.info("Getting SenderCode from apikey={}", apiKey);
```
Comment (GiovanniMancini) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/RestControllerImpl.java]: @and-mora Maybe we could use "Getting SenderCode from internal Id"

**Review by GiovanniMancini — APPROVED** [review]

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/DefaultRestController.java (hunk 1 above)
Comment (and-mora) [reply src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/DefaultRestController.java]: You're right! Thanks for the advice.

File: pom.xml (hunk 2 above)
Comment (and-mora) [reply pom.xml]: At the moment noone, I'm going to exclude the actuator dependency in this PR. We can think to re-introduce it later if needed.

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/RestControllerImpl.java (hunk 3 above)
Comment (and-mora) [reply src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/controller/RestControllerImpl.java]: Done, thank you!

**Pull request merged**